R/data-combined.R

```diff
@@ -0,0 +1,867 @@
+    #' A method to extract a dataframe of simulated and/or observed data
+    #' (depending on instances of which classes have been added to the object).
+    #'
+    #' Note that the order in which you enter different object doesn't matter
+    #' because the returned dataframe is arranged alphabetically by dataset
+    #' name.
+    #'
+    #' @return
+    #'
+    #' In the returned dataframe, the following columns will always be present:
+    #'
+    #' name - group - dataType - xValues - xDimension - xUnit - yValues -
+    #' yErrorValues - yDimension - yUnit - yErrorType - yErrorUnit - molWeight
+
+
+    toDataFrame = function() {
+      # It is deliberate that this "cleaning" happens every time the user calls
+      # this method. R6 has a reference semantics, i.e., the object will be
+      # modified in place and the combined data will be updated with it. But,
+      # for future updates, we do need to retain some information that is
+      # omitted during cleaning.
+      #
+      # For example, we remove offset and scale factor columns during cleaning,
+      # but they need to be retained in the private copy of the combined
+      # dataframe because the datasets that might be added in the future will
+      # have their own offsets and scale factors, which will need to be appended
+      # to the existing ones to be returned by the `dataTransformations()`
+      # active binding. If we were to clean the dataframe in place, this will
+      # not be possible since there won't be offset and scale factors columns to
+      # append to.
+      private$.cleanDF(private$.dataCombined)
+    },
+
+    ## print method -----------------
+
+    #' @description
+    #' Print the object to the console
+
+    print = function() {
+      # group map contains names of the included datasets and grouping details
+      private$printLine("DataCombined", addTab = FALSE)
+      private$printLine("Datasets and groupings", addTab = FALSE)
+      cat("\n")
+      print(private$.groupMap)
+
+      # for method chaining
+      invisible(self)
+    }
+  ),
+
+  # active bindings ---------------------------------------------------
+
+  active = list(
+
+    #' @field names A vector of unique names of datasets contained in the
+    #'   `DataCombined` class instance.
+
+    # just a way to access whatever was specified
+    names = function(value) {
+      if (missing(value)) {
+        private$.names
+      } else {
+        stop(messages$errorPropertyReadOnly(
+          "names",
+          optionalMessage = "Names are assigned using `names` argument in `$addSimulationResults()` or `$addDataSets()` methods."
+        ))
+      }
+    },
+
+    #' @field groupMap A dataframe specifying which datasets have been grouped
+    #'   together and the name and the nature (observed or simulated?) of the
+    #'   data. If a dataset was not assigned to any group, this is denoted by
+    #'   `NA` in the dataframe.
+
+    # just a way to access whatever was specified
```

Comment:
lol.. what?

Reply:
Relic of something that was supposed to be removed. Removed now.